src/components/Web3ReactManager/index.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react'
+import { useWeb3React } from '@web3-react/core'
+
+import useEagerConnect from 'hooks/useEagerConnect'
+import useInactiveListener from 'hooks/useInactiveListener'
+import PageLoader from 'components/PageLoader'
+import { network } from 'connectors'
+
+export default function Web3ReactManager({ children }: { children: JSX.Element }) {
+  const { active, error, activate } = useWeb3React()
+
+  // try to eagerly connect to an injected provider, if it exists and has granted access already
+  const triedEager = useEagerConnect()
+
+  // after eagerly trying injected, if the network connect ever isn't active or in an error state, activate itd
+  useEffect(() => {
+    if (triedEager && !error && !active) {
+      activate(network)
+    }
+  }, [triedEager, error, activate, active])
+
+  // when there's no account connected, react to logins (broadly speaking) on the injected provider, if it exists
+  useInactiveListener(!triedEager)
+
+  // handle delayed loader state
+  const [showLoader, setShowLoader] = useState(false)
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      setShowLoader(true)
+    }, 600)
+
+    return () => {
+      clearTimeout(timeout)
+    }
+  }, [])
+
+  // on page load, do nothing until we've tried to connect to the injected connector
+  if (!triedEager) {
+    return null
+  }
```

Comment:
I am not sure this is the best approach. The user will see a blank white page.

Reply:
This case could be removed as we always try with injected provider when users unable to connect wallet